netket/optimizer/qgt/qgt_jacobian_dense.py

```diff
@@ -94,7 +94,7 @@ def _unscaled_matmul(self, vec: jnp.ndarray) -> jnp.ndarray:
             + self.diag_shift * vec
         )
 
-    @jax.jit
+    @partial(jax.jit, static_argnums=1)
```

Comment:
We can't jit over a function...

Reply:
no, but we can jit over a partial/HashablePartial/jax.Partial.
I think we should add some logic in the base `.solve()` to check if the object is washable, and if not wrap it in `jax.Partial`

Reply:
The thing you wrote won't allow to wrap in arguments that should not be hashable and avoid rejitting.
Arguably it's a bit a mess: there is `jax.Partial` that hashes only the function and everything else must be `jnp.ndarray`, our `nk.HashablePartial` which hashes everything. It would be nice to have something where you can declare what to hash and what not to...

I need to think a bit if we need some flexibility here.But I'd like to allow it...

cc @femtobit

Reply:
In practice this would only be called from the `Vmc` driver which is supplied with a particular `partial` at the beginning, so not much performance is lost over making it static...

Reply:
Sorry, I don't agree.
I routinely find myself playing with modifying the tolerance from a small value at the beginning of the optimisation to a larger value later.
I use a `jax.Partial` to do this. And I am glad it does not recompile at every step...

Reply:
Ah I haven't realised that's possible!  
In any case, I'll rely on you guys to help me write this, I still don't really get much of the more elaborate jax stuff...

Reply:
yes don't worry.
I just need to think a bit about it

Reply:
I think you can remove the explicit `static` argnum now